adapters/sspBC/sspbc.go

```diff
@@ -0,0 +1,406 @@
+	PublisherId string `json:"pubid"`
+	SiteId      string `json:"siteid"`
+	SlotId      string `json:"slotid"`
+}
+
+type adapter struct {
+	endpoint       string
+	bannerTemplate *template.Template
+}
+
+// ---------------ADAPTER INTERFACE------------------
+// Builder builds a new instance of the sspBC adapter
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	// HTML template used to create banner ads
+	const bannerHTML = `<html><head><title></title><meta charset="UTF-8"><meta name="viewport" content="` +
+		`width=device-width, initial-scale=1.0"><style> body { background-color: transparent; margin: 0;` +
+		` padding: 0; }</style><script> window.rekid = {{.SiteId}}; window.slot = {{.SlotId}}; window.ad` +
+		`label = '{{.AdLabel}}'; window.pubid = '{{.PubId}}'; window.wp_sn = 'sspbc_go'; window.page = '` +
+		`{{.Page}}'; window.ref = '{{.Referer}}'; window.mcad = JSON.parse(atob('{{.McAd}}'));</script><` +
+		`/head><body><div id="c"></div><script async crossorigin nomodule src="//std.wpcdn.pl/wpjslib/wp` +
+		`jslib-inline.js" id="wpjslib"></script><script async crossorigin type="module" src="//std.wpcdn` +
+		`.pl/wpjslib6/wpjslib-inline.js" id="wpjslib6"></script></body></html>`
+
+	bannerTemplate, err := template.New("banner").Parse(bannerHTML)
+	if err != nil {
+		return nil, err
+	}
+
+	bidder := &adapter{
+		endpoint:       config.Endpoint,
+		bannerTemplate: bannerTemplate,
+	}
+
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	formattedRequest, err := formatSspBcRequest(a, request)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	requestJSON, err := json.Marshal(formattedRequest)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	requestURL, err := url.Parse(a.endpoint)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	// add query parameters to request
+	queryParams := requestURL.Query()
+	queryParams.Add("bdver", version) // adapter version
+	queryParams.Add("inver", "0")     // integration version (adapter, tag, ...)
+	requestURL.RawQuery = queryParams.Encode()
+
+	requestData := &adapters.RequestData{
+		Method: http.MethodPost,
+		Uri:    requestURL.String(),
+		Body:   requestJSON,
+	}
+
+	return []*adapters.RequestData{requestData}, nil
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, externalResponse *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if externalResponse.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if externalResponse.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d.", externalResponse.StatusCode),
+		}
+		return nil, []error{err}
+	}
+
+	// unmarshall original request - we will need it to recover imp.Id
+	var requestBody openrtb2.BidRequest
```

Comment:
I don't believe that you have to unmarshal externalRequst.Body. The internalRequest is the same pointer that MakeBids() was given, so when you moved Imp.ID to Imp.TagID, that change is still there in the internalRequest you have now.

Reply:
You are right. I fixed it. Thanks.